### DISCUSSION TITLE
Should GitHub Copilot be free?

### DISCUSSION THREAD

**Discussion by ollie-iterators** [community/community#20284] [Other Feature Feedback, Questions, & Ideas]
GitHub Copilot used open source code for the training of the product. That means that it should be free for people to use. It would also allow the most amount of people to use it to make their code better and thus help with the training.

**Comment by redegg89** [comment]
On one hand, the code is free and some of the licenses definitely wouldn't like using the code for AI training. On the other hand, it's not free to run the servers that power the AI in the first place. While I wouldn't pay for it, Github has to be profitable somehow.

**Reply by SebGrd** [reply]
i was thinking the same.
It should be free for every GitHub contributors since their code is the base data for Copilot. And companies should have to pay to use it.
It seems to be the best of both world in my opinion.
What do you guys think?

**Reply by redegg89** [reply]
@SebGrd Sounds like a good idea to me, but then you run into the problem of "well technically we're contributors too," since a lot of companies also have source code on github. I'm thinking maybe more of only having to pay for commercial use, since companies would still have to pay (I think).

**Reply by MulleDK19** [reply]
> On one hand, the code is free and some of the licenses definitely wouldn't like using the code for AI training. On the other hand, it's not free to run the servers that power the AI in the first place. While I wouldn't pay for it, Github has to be profitable somehow.

The license on repositories don't matter. When you use GitHub, you agree to the terms of service, which specifically says you grant GitHub the right to use your code to make or improve GitHub products. That's why it's named GitHub Copilot instead of a separate product name. By making it part of GitHub, they're legally allowed to use your code for making or improving their service, which Copilot is/does.

But I still think it should be free. Or a lot cheaper. $10/mo is a lot for me, especially since Copilot isn't the only subscription I have..

**Comment by buttonfly1000** [comment]
In the end, it will be free

**Reply by martin-braun** [reply]
Please, tell me more.

**Reply by Hellzbellz123** [reply]
everyone and thier dog saw tabnine/copilot and their will be free alternatives, tabnine already has a free tier even if its crap

**Comment by Babaray88** [comment]
Yes

**Comment by Babaray88** [comment]
Good

**Comment by Lexian-droid** [comment]
The problem lies on the fact that copilot is trained on Github, We all uploaded stuff to github, I presume, I uploaded my fair share of crap to it, even forked a few repositories, so considering that, Copilot most definitely pulled code from our repositories as well, not to mention, it learns from what you're doing, which means it is actively using your code as well.

Once the dang thing autofilled an imgur link, you know what it was? a picture of 2 guys making out masturbating. I reported this, and it hasn't happened again sense, but it goes to show, if you comment shit in, theres no telling if copilot isn't going to share that stuff, some coders put their login credentials to certain things into documents, say copilot reads them, and it thinks "Well maybe someone might want to use these" well suddenly, any other programmer will see your login credentials.

It's flawed, and honestly I don't like how we are paying for something they effectively used us to make. theres a solution here, but it's not the right solution, copilot steals code any way it feels, it's literally if stack overflow was an AI. Considering the fact that github went free, it's their own fault that they can't afford to make it free, but it's also backed by a multi-billion dollar company (Microsoft) ya'll ever heard of Perspective API? It's an API built by google, 100% free, no strings attached.

Why is it that Google can create a completely automated text processing AI that detects toxicity, and Microsoft can't make copilot free?
At least make it free for the average consumer, seriously, how much resources are regular users going to hog up? charge companies, but make it free for consumers, like normal companies do. Something like this that is built by the community, should be for the community. and thus; free.

**Reply by librity** [reply]
If it's free YOU are the product.
The best we can do to get back at microsoft is upload terabytes of pictures to a repo and fork it 500 times.

**Reply by Lexian-droid** [reply]
I hope you realize, we already are the product, Github uses our code to feed the bot, they admitted it, it's in their TOS.
Personally, It should be free seeing as again, we are the product. We already are.

**Reply by librity** [reply]
> I hope you realize, we already are the product, Github uses our code to feed the bot, they admitted it, it's in their TOS.

Gee, I never thought of it like that

**Comment by ollie-iterators** [comment]
OpenAI is reducing the price of GPT-3 (I know that Copilot is built on OpenAI Codex, which is a direct descendant of GPT-3), but the price change should impact the price of Codex, which should make it easier for GitHub to make it free.

**Reply by ollie-iterators** [reply]
https://help.openai.com/en/articles/6485334-openai-api-pricing-update-faq. GPT-3's price has just been reduced by 2/3's

**Comment by Babaray88** [comment]
Ok

**Reply by ARMMWORXS** [reply]
Thanks.  JAR.
